gcp/api/cloudbuild.yaml

```diff
@@ -0,0 +1,45 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# api test runner
+# This should be triggered on changes to the following:
+# gcp/api/poetry.lock
+# gcp/api/run_tests.sh
+# gcp/api/*.py
+# gcp/api/**/*.py
+# osv/*.py
+# osv/**/*.py
+
+steps:
+- name: 'gcr.io/cloud-builders/git'
+  id: 'init'
+  args: ['submodule', 'update', '--init']
+  # if this is invoked from another cloud build, this will fail as it is not a git repo
+  # the invoking cloud build file should run this step.
+  allowFailure: true
+- name: 'gcr.io/oss-vdb/ci'
+  id: 'sync'
+  dir: gcp/api
+  args: ['poetry', 'sync']
+
+- name: 'gcr.io/oss-vdb/ci'
+  id: 'api-tests'
+  dir: gcp/api
+  #TODO: Update test scripts to support not supplying a credential.
+  args: ['bash', '-ex', 'run_tests.sh', '/workspace/dummy.json']
+  env:
+    - CLOUDBUILD=1
+  waitFor: ['init', 'sync']
```

Comment:
nit: is this necessary? without this argument we also have the same desired effect right? 

(similar comment for other instances).

Reply:
eh - it's necessary for workers & lib-tests, since they have multiple test steps that we want to run concurrently.
I don't think there's much harm in also having it in api/website/vulnfeeds just to be consistent.

Reply:
I also meant to put `waitFor: ['-']` in the `sync` steps but I forgot...